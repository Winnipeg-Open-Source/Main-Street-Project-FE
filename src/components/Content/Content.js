import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex } from 'pcln-design-system';

const StyledContent = styled(Flex)`
    flex: 1;
    height: 100%;
    overflow-y: scroll;
`;

function Content ({ className, children }) {
    return (
        <StyledContent className={className} color='primary.light'>
            {children}
        </StyledContent>
    );
}

Content.displayName = 'Content';

Content.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
};

Content.defaultProps = {
    className: '',
};

export default Content;
