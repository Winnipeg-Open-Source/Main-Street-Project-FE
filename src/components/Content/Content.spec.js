import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import Content from './Content';

describe ('Content', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<Content />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('renders content successfully', () => {
        const { getByText } = renderWithTheme(<Content>Content</Content>);

        expect(getByText(/Content/i)).toBeInTheDocument();
    });
});
