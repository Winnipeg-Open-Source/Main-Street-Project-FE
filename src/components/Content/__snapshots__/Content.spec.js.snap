// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-LzLMl bRtbEd sc-LzLMm jaRkvP"
  >
    <div
      class="sc-AykKC sc-AykKE sc-LzLMn jfEarE"
      color="primary.light"
    />
  </div>
</DocumentFragment>
`;
